presto-hive/src/test/java/com/facebook/presto/hive/s3/TestPrestoS3FileSystem.java

```diff
@@ -794,6 +806,191 @@ public ObjectListing listObjects(ListObjectsRequest listObjectsRequest)
         }
     }
 
+    @Test(dataProvider = "streamingConfigs")
+    public void testMultipart(boolean streaming)
+            throws IOException, URISyntaxException
+    {
+        try (PrestoS3FileSystem fs = new PrestoS3FileSystem()) {
+            MockAmazonS3 s3 = new MockAmazonS3();
+            Configuration conf = new Configuration();
+            conf.setLong(S3_MULTIPART_MIN_PART_SIZE, 512L);
+            conf.setBoolean(S3_USE_STREAMING_UPLOADS, streaming);
+            fs.initialize(new URI("s3n://test-bucket/"), conf);
+            fs.setS3Client(s3);
+            try (FSDataOutputStream out = fs.create(new Path("test"))) {
+                out.flush();
+            }
+            // since we didn't exceed the part size, no multipart requests should be sent
+            assertEquals(s3.initiateMultipartRequests, 0);
+            assertEquals(s3.uploadPartRequests, 0);
+            assertEquals(s3.completePartUploadRequests, 0);
+            assertEquals(s3.putObjectRequests, 1);
+
+            try (FSDataOutputStream out = fs.create(new Path("test"))) {
+                out.write(new byte[512]);
+            }
+            // same as above
+            assertEquals(s3.initiateMultipartRequests, 0);
+            assertEquals(s3.uploadPartRequests, 0);
+            assertEquals(s3.completePartUploadRequests, 0);
+            assertEquals(s3.putObjectRequests, 2);
+
+            try (FSDataOutputStream out = fs.create(new Path("test"))) {
+                out.write(new byte[600]);
+                out.write(new byte[600]);
+            }
+            assertEquals(s3.initiateMultipartRequests, streaming ? 1 : 0);
+            assertEquals(s3.uploadPartRequests, streaming ? 2 : 0);
+            assertEquals(s3.completePartUploadRequests, streaming ? 1 : 0);
+            assertEquals(s3.putObjectRequests, streaming ? 2 : 3);
+        }
+    }
+
+    private void assertDirEmpty(java.nio.file.Path path)
+            throws IOException
+    {
+        try (Stream<java.nio.file.Path> dirIter = Files.walk(path)) {
```

Comment:
This is more complex than it has to be. You can just do path.toFile().list()

Reply:
Thanks for the pointer, fixed